pkg/sypgp/sypgp.go

```diff
@@ -1014,38 +1018,35 @@ func (keyring *Handle) importPrivateKey(entity *openpgp.Entity) error {
 		return &KeyExistsError{fingerprint: entity.PrivateKey.Fingerprint}
 	}
 
-	// Check if the key is encrypted, if it is, decrypt it
-	if entity.PrivateKey == nil {
-		return fmt.Errorf("corrupted key, unable to recover data")
-	}
-
-	// Make a clone of the entity
-	newEntity := &openpgp.Entity{
-		PrimaryKey:  entity.PrimaryKey,
-		PrivateKey:  entity.PrivateKey,
-		Identities:  entity.Identities,
-		Revocations: entity.Revocations,
-		Subkeys:     entity.Subkeys,
-	}
+	newEntity := *entity
 
+	var password string
 	if entity.PrivateKey.Encrypted {
-		if err := DecryptKey(newEntity, "Enter your old password : "); err != nil {
+		password, err = interactive.AskQuestionNoEcho("Enter your key password : ")
+		if err != nil {
+			return err
+		}
+		if err := newEntity.PrivateKey.Decrypt([]byte(password)); err != nil {
 			return err
 		}
 	}
 
-	// Get a new password for the key
-	newPass, err := interactive.GetPassphrase("Enter a new password for this key : ", 3)
-	if err != nil {
-		return err
+	if setNewPassword {
+		// Get a new password for the key
+		password, err = interactive.GetPassphrase("Enter a new password for this key : ", 3)
+		if err != nil {
+			return err
+		}
 	}
 
-	if err := EncryptKey(newEntity, newPass); err != nil {
-		return err
+	if password != "" {
+		if err := newEntity.PrivateKey.Encrypt([]byte(password)); err != nil {
+			return err
+		}
 	}
 
 	// Store the private key
-	if err := keyring.appendPrivateKey(newEntity); err != nil {
+	if err := keyring.appendPrivateKey(&newEntity); err != nil {
```

Comment:
Why is the `&` here? Is it also necessary for the other calls to `keyring.appendPubKey`, and `keyring.appendPrivateKey`?

Reply:
@WestleyK `&` is a pointer gian operation. It is there b/c `keyring.appendPubKey` accepts pointer, not value.